packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -188,25 +194,33 @@ export class HbarLimitService implements IHbarLimitService {
 
     const spendingLimit = HbarLimitService.TIER_LIMITS[spendingPlan.subscriptionTier].toTinybars();
 
+    // note: estimatedTxFee is only applicable in a few cases (currently, only for file transactions).
+    //      In most situations, estimatedTxFee is set to 0 (i.e., not considered).
+    //      In such cases, it should still be true if spendingPlan.amountSpent === spendingLimit.
     const exceedsLimit =
       spendingLimit.lte(spendingPlan.amountSpent) || spendingLimit.lt(spendingPlan.amountSpent + estimatedTxFee);
+
     this.logger.trace(
-      `${requestDetails.formattedRequestId} ${user} ${exceedsLimit ? 'should' : 'should not'} be limited: amountSpent=${
+      `${requestDetails.formattedRequestId} User ${
+        exceedsLimit ? 'has' : 'has NOT'
+      } exceeded HBAR rate limit threshold: user=${user}, amountSpent=${
         spendingPlan.amountSpent
-      }, estimatedTxFee=${estimatedTxFee} tℏ, spendingLimit=${spendingLimit.toString()} tℏ`,
+      }, estimatedTxFee=${estimatedTxFee}, spendingLimit=${spendingLimit}, txConstructorName=${txConstructorName}, mode=${mode}, methodName=${methodName}`,
     );
+
     return exceedsLimit;
   }
 
   /**
-   * Add expense to the remaining budget.
+   * Add expense to the remaining budget and update the spending plan if applicable.
    * @param {number} cost - The cost of the expense.
    * @param {string} ethAddress - The Ethereum address to add the expense to.
    * @param {RequestDetails} requestDetails The request details for logging and tracking.
    * @returns {Promise<void>} - A promise that resolves when the expense has been added.
    */
   async addExpense(cost: number, ethAddress: string, requestDetails: RequestDetails): Promise<void> {
     const ipAddress = requestDetails.ipAddress;
+
     if (!ethAddress && !ipAddress) {
       throw new Error('Cannot add expense without an eth address or ip address');
     }
```

Comment:
We can move the logic for the `totalLimit` above this check (similar to `shouldLimit`) to make sure that expenses are added to the `totalBudget` when both `ethAddress` and `ipAddress` are missing.
```suggestion
    const newRemainingBudget = this.remainingBudget.toTinybars().sub(cost);
    this.remainingBudget = Hbar.fromTinybars(newRemainingBudget);
    this.hbarLimitRemainingGauge.set(newRemainingBudget.toNumber());
    
    if (!ethAddress && !ipAddress) {
      this.logger.warn(`${requestDetails.formattedRequestId} Cannot add expense to an HbarSpendingPlan without a passed ETH or IP address`);
      return;
    }
```

Reply:
Good point! Updated